src/connector/src/parser/json_parser.rs

```diff
@@ -102,7 +128,12 @@ impl JsonParser {
                 change_event_op,
             )
         } else {
-            let value = simd_json::to_borrowed_value(&mut payload)
+            if payload.is_none() {
```

Comment:
Performance is regressed because of this additional check
```
json_parser             time:   [473.68 ms 474.11 ms 474.55 ms]
                        change: [+1.5506% +1.6483% +1.7479%] (p = 0.00 < 0.05)
```